src/OpenTelemetry.ResourceDetectors.Container/K8sMetadataFetcher.cs

```diff
@@ -0,0 +1,76 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+using System.IO;
+using System.Text;
+
+namespace OpenTelemetry.ResourceDetectors.Container;
+
+internal sealed class K8sMetadataFetcher : IK8sMetadataFetcher
+{
+    private const string KubernetesServiceHostKey = "KUBERNETES_SERVICE_HOST";
+    private const string KubernetesServicePortKey = "KUBERNETES_SERVICE_PORT_HTTPS";
+    private const string KubernetesHostnameKey = "HOSTNAME";
+    private const string KubernetesPodNameKey = "KUBERNETES_POD_NAME";
+    private const string KubernetesContainerNameKey = "KUBERNETES_CONTAINER_NAME";
+    private const string KubernetesNamespacePath = "/var/run/secrets/kubernetes.io/serviceaccount/namespace";
```

Comment:
One more small edge case (we can address it in follow-up PR, as it would require some preparation for testing.
CGroup based resolution was Linux-specific, but K8s mechanics should work in Windows also.
We should test, where K8s places namespace/token/certificate on Windows pods. Most probably, it would be the same path on drive `C:`. In that case, we should use C-drive prefix on Windows, otherwise it would not work when current working directory is on another drive.